src/components/CategoryPicker.tsx

```diff
@@ -1,21 +1,34 @@
-import lodashGet from 'lodash/get';
-import React, {useMemo, useState} from 'react';
+import {useMemo, useState} from 'react';
 import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
-import OptionsSelector from '@components/OptionsSelector';
+import type {OnyxEntry} from 'react-native-onyx';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps, propTypes} from './categoryPickerPropTypes';
+import type * as OnyxTypes from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import OptionsSelector from './OptionsSelector';
 
-function CategoryPicker({selectedCategory, policyCategories, policyRecentlyUsedCategories, onSubmit}) {
+type CategoryPickerOnyxProps = {
+    policyCategories: OnyxEntry<OnyxTypes.PolicyCategories>;
+    policyRecentlyUsedCategories: OnyxEntry<OnyxTypes.RecentlyUsedCategories>;
+};
+
+type CategoryPickerProps = CategoryPickerOnyxProps & {
+    /** It's used by withOnyx HOC */
+    // eslint-disable-next-line react/no-unused-prop-types
+    policyID: string;
+    selectedCategory: string;
+    onSubmit: () => void; // TO DO: Check if optional
```

Comment:
I think it should be required 👍

Reply:
When I look at the usage, it should have an argument:
```ts
    onSubmit: (category: Category) => void;
```

Reply:
Good catch! But I'm not sure about this type. 
We have the following code in the component:
```
const selectedOptions = useMemo(() => {
        if (!selectedCategory) {
            return [];
        }

        return [
            {
                name: selectedCategory,
                enabled: true,
                accountID: null,
            },
        ];
    }, [selectedCategory]);
```

So it looks like `Category` defines 2 of 3 keys used here 🤔

Reply:
Ah sorry it was a pseudo code, let's create a type that makes more sense

Reply:
After some digging I noticed that we don't need this `accountID` there. Whenever we treat `selectedOptions` like `Options` in `OptionsListUtils` omitting this `accountID: null` doesn't change anything. Also, I tested it and works fine.

Reply:
Makes sense 👍